### PR TITLE
feat: create kics plugin

### PR DESCRIPTION
**Pull request by otaviomacedo**
Some things I'd like to call attention to:

- The main logic is in the `plugin.ts` file, where `KicsValidator` implements the plugin interface. It calls the correct `kics` binary based on the operating system and hardware architecture, passing the template it receives from CDK. If this operation succeeds, it reads the results from the output file and reports back to the framework.
- The CDK API has lots of names with a `Beta1` suffix. This is because the `core` module is considered stable, but the policy validation feature, specifically, is still experimental. This is the strategy we use for these cases. When it goes GA, the suffixes will be removed
- The `--fail-on`, `--exclude-queries`, `--exclude-categories` and `--exclude-severities` CLI options are exposed by the plugin as constructor properties.
- Many files are auto-generated by [projen](https://projen.io/). You can configure how it works by tweaking properties and calling methods in the project object.
- You will need to set up some secrets in the repository:
	- `PROJEN_GITHUB_TOKEN`: used to create and auto-approve PRs. Projen uses this to keep the dependencies up-to-date.
	- `NPM_TOKEN`, when you are ready to publish to NPM. To enable this, set `release: true` in the projen project.
- This PR does not include any configuration for jsii to generate bindings in other languages. This should probably be a separate discussion. Nothing blocks you from releasing to NPM initially and then add more languages as you are ready.
- There are two jobs related to the `kics` binary:
	- `projenrc/update-kics.ts`: reads the latest version from the GitHub releases and updates a file called `kics-version.json`. Runs once a week.
	- `projenrc/bundle-kics.ts`: clones and builds (using `goreleaser`) `kics` for different combinations of operating systems and hardware architectures. Uses the `kics-version.json` file mentioned above. Runs on every build.

To get started:

- Install Node.js and npm: https://docs.npmjs.com/downloading-and-installing-node-js-and-npm
- Install yarn: https://yarnpkg.com/getting-started/install
- To run the tests, in the project root: `npx projen test`.
- To run a CDK synthesis, end-to-end: `cdk -a "ts-node app.ts" synth --debug`, where `app.ts` is a CDK app written in TypeScript, such as:

```ts
import { App, aws_s3 as s3, Stack } from 'aws-cdk-lib';
import { KicsValidator } from '../src/plugin';

const app = new App({
  policyValidationBeta1: [
    new KicsValidator({
    }),
  ],
});

// WHEN
const stack = new Stack(app, 'Stack');
new s3.Bucket(stack, 'Bucket');

app.synth();
```

### CODE REVIEW

File: .projenrc.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { JsiiProject } from 'projen/lib/cdk';
+import { BundleKics } from './projenrc';
+
+const project = new JsiiProject({
+  author: 'Checkmarx',
+  authorAddress: 'REPLACEME@checkmarx.com',
```
Comment (otaviomacedo) [inline .projenrc.ts]: Replace this with a real address.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,122 @@
-# kics-cdk-validator-plugin
-A KICS plugin for AWS CDK
+# CDK Kics Validator Plugin
+
+<!--BEGIN STABILITY BANNER-->
+
+---
+
+![cdk-constructs: Experimental](https://img.shields.io/badge/cdk--plugin-experimental-important.svg?style=for-the-badge)
```
Comment (otaviomacedo) [inline README.md:7]: I'm not sure what title you want to actually use in this banner. Set it provisonally as "CDK-PLUGIN | EXPERIMENTAL", but feel free to change this.

File: .projenrc.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    '@octokit/types',
+    '@octokit/rest',
+    'mock-fs',
+    '@types/mock-fs',
+    'fs-extra',
+    '@types/fs-extra',
+  ],
+  name: 'kics-cdk-validator-plugin',
+  projenrcTs: true,
+  release: false,
```
Comment (otaviomacedo) [inline .projenrc.ts]: Flipping this to true will make projen generate a new workflow file, with actions to pulish releases to GitHub and to NPM.

**Comment by kaplanlior** [comment]
Thanks @otaviomacedo

**Pull request merged**